src/main/java/website/frontrow/game/Game.java

```diff
@@ -49,29 +45,35 @@ public Game(ArrayList<Level> levels, ArrayList<Player> players)
      */
     public void tick()
     {
-        if(running)
+        if (running)
         {
-            if(keyListener != null)
+            if (keyListener != null)
             {
                 keyListener.update();
             }
             currentLevel.tick();
-
-            if(currentLevel.getEnemies() == 0)
-            {
-                levelWon();
-            }
         }
         updateObservers();
     }
 
     /**
+     * Replaces the current level with a copy of the level at the current index.
+     */
+    private void loadCurrentLevel()
+    {
+        this.currentLevel = levelPack.get(currentIndex).duplicate();
+        if(currentLevel != null)
+        {
+            currentLevel.addObserver(this);
```

Comment:
The level loses all the references to it and will get cleaned up by the garbage collector.
No need to explicitly unregister.